packs/democratic_csi_nfs/variables.hcl

```diff
@@ -0,0 +1,120 @@
+variable "job_name" {
+  description = "The prefix to use as the job name for the plugins (ex. democratic_csi_controller for the controller)."
+  type        = string
+  default     = "democratic_csi"
+}
+
+variable "datacenters" {
+  description = "A list of datacenters in the region which are eligible for task placement."
+  type        = list(string)
+  default     = ["dc1"]
+}
+
+variable "region" {
+  description = "The region where the job should be placed."
+  type        = string
+  default     = "global"
+}
+
+variable "plugin_id" {
+  description = "The ID to register in Nomad for the plugin."
+  type        = string
+  default     = "org.democratic-csi.nfs"
+}
+
+variable "plugin_namespace" {
+  description = "The namespace for the plugin job."
+  type        = string
+  default     = "default"
+}
+
+variable "plugin_image" {
+  description = "The container image for democratic-csi."
+  type        = string
+  default     = "docker.io/democraticcsi/democratic-csi:latest"
+}
+
+variable "plugin_csi_spec_version" {
+  description = "The CSI spec version that democratic-csi will comply with."
+  type        = string
+  default     = "1.5.0"
+}
+
+variable "plugin_log_level" {
+  description = "The log level for the plugin."
+  type        = string
+  default     = "debug"
+}
+
+variable "nfs_share_host" {
+  description = "The IP address of the host for the NFS share."
+  type        = string
+}
+
+variable "nfs_share_base_path" {
+  description = "The base directory exported from the NFS share host."
+  type        = string
+}
+
+variable "nfs_controller_mount_path" {
+  description = "The path where the NFS mount is mounted as a host volume for the controller plugin."
+  type        = string
+}
+
+variable "nfs_dir_permissions_mode" {
+  description = "The unix file permissions mode for the created volumes"
+  type        = string
+  default     = "0777"
+}
+
+variable "nfs_dir_permissions_user" {
+  description = "The unix user that owns the created volumes."
+  type        = string
+  default     = "root"
+}
+
+variable "nfs_dir_permissions_group" {
+  description = "The unix group that owns the created volumes."
+  type        = string
+  default     = "root"
+}
```

Comment:
All this stuff gets stuffed into the YAML blob for the plugin's config file. We could have these all as fields of an object or we could just have a freeform config file blog for the user (but that seems more error prone)?

Reply:
Fields of an object is actually an antipattern (IMO) because you cant overwrite single values with HLC2 parsing. Which means you have to copy over an entire object in your vars file just to get a single field to change.

Between this and freeform config, I don't have a strong preference. I think there's just a point where if you have like tons of variables and conditionals to write a config file, then you might as well just give them a full file variable.  5 variables seems okay to keep in HCL, IMO, but is nearing the point where I'd switch to raw file. Where to draw the line is a taste issue probably.

Reply:
The big advantage to explicit fields would be validation on ingest, but that too is probably more a matter of personal style.

Reply:
👍